spec/tracking_system_spec.rb

```diff
@@ -0,0 +1,252 @@
+    end
+
+    it "returns a list" do
+      expect(@tracker.view_all_rooms).must_be_kind_of Array
+    end
+
+    it 'must contain 20 elements' do
+      expect(@tracker.view_all_rooms.length).must_equal 20
+    end
+
+    it 'must contain only instances of Room' do
+      expect(@tracker.view_all_rooms.all? {|room| room.class == Room}).must_equal true
+    end
+  end
+
+
+  describe "#view_reservations_on(date)" do
+    before do
+      @tracker = TrackingSystem.new
+    end
+
+    it "returns an array" do
+      @reservation = @tracker.add_reservation(start_time: Date.new(2018,10,5), end_time: Date.new(2018,10,10), number_of_rooms:1)
+      @reservation1 = @tracker.add_reservation(start_time: Date.new(2018,10,5), end_time: Date.new(2018,10,10), number_of_rooms:1)
+      expect(@tracker.view_reservations_on(Date.new(2018,10,5))).must_be_kind_of Array
+    end
+
+    it "raises an ArgumentError if no reservations are available on given date" do
+      expect{@tracker.view_reservations_on(Date.new(3333,10,5))}.must_raise ArgumentError
+    end
+
+#
+    it "raises an ArgumentError if given date is not an instance of Date class" do
+      expect{@tracker.view_reservations_on("some date")}.must_raise ArgumentError
+    end
+  end
+
+
+
+  describe "#total_cost_of_reservation" do
+    before do
+      @tracker = TrackingSystem.new
+      attributes = {room_num: 1 ,start_time: Date.new(2018,8,1),end_time: Date.new(2018,9,1),price: 200.00}
+      @reservation = Reservation.new(attributes)
+    end
+
+    it "raises ArgumentError unless argument is an instance of Reservation" do
+      expect{@tracker.total_cost_of_reservation("hello there!")}.must_raise ArgumentError
+    end
+
+    it "returns an instance of a Float" do
+      expect(@tracker.total_cost_of_reservation(@reservation)).must_be_kind_of Float
+    end
+
+    it "returns a postive number" do
+      expect(@tracker.total_cost_of_reservation(@reservation)).must_equal 6200.00
+    end
+  end
+
+
+  describe "#view_available_rooms_on" do
+    before do
+      @tracker = TrackingSystem.new
+    end
+
+    it "raises ArgumentError if start_time is > end_time" do
+      @tracker = TrackingSystem.new
+      expect{@tracker.view_available_rooms_on(start_time: Date.new(2018,10,5),end_time:Date.new(2018,9,5))}.must_raise ArgumentError
+    end
+
+
+    it "returns an array" do
+      @tracker = TrackingSystem.new
+      @available_rooms = @tracker.view_available_rooms_on(start_time: Date.new(2018,8,1),end_time:Date.new(2018,9,5))
+      expect(@available_rooms).must_be_kind_of Array
+    end
+
+    it "returns an array containing instances of Room" do
+      @tracker = TrackingSystem.new
+      @available_rooms = @tracker.view_available_rooms_on(start_time: Date.new(2018,8,1),end_time:Date.new(2018,9,5))
+      expect(@available_rooms[0]).must_be_kind_of Room
+    end
+
+    it "raises ArgumentError if no rooms are available on these dates(aka they're in a block or reserved)" do
```

Comment:
This test assumes too much detail about how `Room` is implemented.  Instead what you could do is add 20 reservations on a given date using `TrackingSystem`' s method.  Then try making a 21st reservation on the same date range.